Add documentation for Logger with Transport Client

dadoonet opened the pull request:
Closes #21331

dadoonet commented:
@jasontedor Could you review this doc please?

jasontedor commented on docs/java-api/index.asciidoc:
```diff
@@ -35,6 +35,70 @@ For example, you can define the latest version in your `pom.xml` file:
 </dependency>
 --------------------------------------------------
 
+=== Log4J2 Logger
```
Nit: Log4J2 -> Log4j 2

jasontedor commented on docs/java-api/index.asciidoc:
```diff
@@ -35,6 +35,70 @@ For example, you can define the latest version in your `pom.xml` file:
 
+=== Log4J2 Logger
+
+You need to also include Log4J2 dependencies:
```
Nit: Log4J2 -> Log4j 2

dadoonet commented on docs/java-api/index.asciidoc:
```diff
@@ -35,6 +35,70 @@ For example, you can define the latest version in your `pom.xml` file:
+<dependency>
+    <groupId>org.apache.logging.log4j</groupId>
+    <artifactId>log4j-api</artifactId>
+    <version>2.6.2</version>
```
Ha right! Thanks a lot. I'll fix that.

jasontedor commented:
Also, it occurred to me that these versions in this doc could easily go out of date when we upgrade the Log4j version in version.properties. Could you add a commit to the line item there for Log4j to upgrade the version in these docs when a change is made to the version there?

dadoonet commented:
@jasontedor I updated based on your comments. Thanks!

> Could you add a commit to the line item there for Log4j to upgrade the version in these docs when a change is made to the version there?

I just added that.

Ideally, as I wrote in https://github.com/elastic/docs/issues/4, I wish we could have a real project with all the java code we added in documentation which is running for real.
From this project, using code snippet "a la maven" would be neat.

dadoonet commented:
Also added 
* 5.0: 6348e54650acba3e0188e38d8905cf037cd722c5 and d91ad4e12f28e1b0e723b6223591464ea5be9b77 (with Log4j 2.6.2 instead of 2.7)
* master (6.0): 5336e72fbb85b4d63c8466f64166b58f26a50bb5
